### ISSUE TITLE
🐛 user-provided comparison function does not correctly implement a total order

### ISSUE THREAD

**Issue by paul-vd** [biomejs/biome#5700]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.13.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.1

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         /projects/dashboard/biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

When I run the `biome check` command I get the following report:

```sh
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: library/core/src/slice/sort/shared/smallsort.rs:865:5
Thread Name: biome::workspace_worker_4
Message: user-provided comparison function does not correctly implement a total order
```

### Expected result

It should not display this message.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
